### DISCUSSION TITLE
What is the new equivalent of @yarnpkg/lockfile? Can I help make one?

### DISCUSSION THREAD

**Discussion by naugtur** [yarnpkg/berry#4491] [Q&A]
I'm using some tools that depend on @yarnpkg/lockfile and it seems it's no longer being developed. Is there a berry-equivalent lockfile parser? 
Looking at the packages folder I couldn't find one. Looks like it's embedded in core?

A standalone parser package with string input and as little dependencies as possible would be super helpful. 
I'm willing to participate in creating that.

**Comment by arcanis** [comment]
It's just standard yaml now. The only special treatment we do is to split top-level keys per commas (so `"a,b": 42` is the same as `{a: 42, b: 42}`). Note however that depending on what your use case is, using the Yarn APIs (similar to what plugins do) might be a better idea than reimplementing them. What do you want to achieve?

**Reply by naugtur** [reply]
My intention is to extend the support in https://github.com/lirantal/lockfile-lint to cover all lockfiles, including for berry which is gaining popularity. Getting all versions supported with one package would be great.

Shaving off some dependencies and reliance on `fs` and `os` would also be useful for trust-nobody supply chain security usecases.

And since you have the code to migrate from old lockfile to new anyway, putting it in the parser package is mostly a matter of someone knowing what to extract and someone else (me?) contributing an API surface and documenting it.

**Comment by merceyz** [comment]
> Looks like it's embedded in core?

The core uses [`@yarnpkg/parsers`](https://github.com/yarnpkg/berry/tree/86cf338ead6c5d030b36e4841ee7e538fe8654c1/packages/yarnpkg-parsers) which parse both new and old lockfiles.

**Reply by naugtur** [reply]
Oh, this might be what I was looking for!

Love the docs on that one :D

**Reply by naugtur** [reply]
Here's a deal:
If someone gives me at least 3 links to it being used as intended so that I don't have to dig through the codebase I'll document  @yarnpkg/parsers in return.

For now it seems to me it's used for parsing yaml and commandline args and who knows what else ;)

**Reply by merceyz** [reply]
[Here](https://github.com/yarnpkg/berry/blob/86cf338ead6c5d030b36e4841ee7e538fe8654c1/packages/yarnpkg-core/sources/Project.ts#L315) it's used for `yarn@>=2` lockfiles and [here ](https://github.com/yarnpkg/berry/blob/1d85482c5f526cea96fbb80cb58adf68eaf33050/packages/yarnpkg-core/sources/LegacyMigrationResolver.ts#L47) for `yarn@1 `

**Reply by naugtur** [reply]
Ok, so it works for parsing correct yarn.lock v1 

But it also doesn't complain if I feed it this
```
{
    "cat-generator": meow
}
```

Is that expected?
